tests/test_zlib.rs

```diff
@@ -0,0 +1,331 @@
+use anyhow::Context;
+use fs_extra::dir::{copy,CopyOptions};
+use serial_test::serial;  // for tests that change directories
+use std::fs::File;
+use std::io::Read;
+use std::path::{PathBuf, Path};
+use tempfile::{TempDir,tempdir};
+use xshell::{Cmd, pushd};
+
+use bom;
+use bom::bom::options::{BitcodeOptions,ExtractOptions};
+
+mod common;
+
+/// Runs a series of tests by using/building the zlib sources.  The zlib
+/// distribution is rather ideal for this because:
+///
+///  1. It is relatively small, but does consist of multiple sources
+///  2. It builds both shared and static libraries
+///  3. It builds sample executables, some statically linked, some share linked
+///
+/// These features allow a number of different verifications of build-bom to be
+/// performed here.
+
+// Download the file at the given URL to [tests/sources/<filename>] if it
+// doesn't already exist, returning the canonical absolute path to that file.
+fn fetch_if_needed(url : &str, filename : &str) -> anyhow::Result<PathBuf> {
+    let path = Path::new(common::SOURCE_DIR).join(filename);
+    if !path.exists() {
+        let cmd = Cmd::new("wget").arg("-O").arg(path.as_path()).arg(url);
+        cmd.run()?;
+    }
+
+    let abs_path = std::fs::canonicalize(path.as_path())?;
+    Ok(abs_path)
+}
+
+
+#[allow(dead_code)]  // tdir is holding a resource, but never directly used
+struct ZlibBld {
+    tdir: TempDir,
+    tgt_path: PathBuf,
+    zlib_version: String,
+}
+
+lazy_static::lazy_static! {
+    static ref ZLIB_BLD: ZlibBld = {
+        match zlib_do_build() {
+            Ok((tdir, zlib_version, tgt_path)) =>
+                ZlibBld { tdir, tgt_path, zlib_version },
+            Err(e) =>
+                panic!("Unable to compile zlib for testing: {:?}", e),
+        }
+    };
+}
+
+
+fn zlib_do_build() -> anyhow::Result<(TempDir, String, PathBuf)> {
+    let url = "https://www.zlib.net/current/zlib.tar.gz";
+    let filename = "zlib.tar.gz";
+    let abs_src = fetch_if_needed(url, filename)?;
```

Comment:
This can be a different PR, but I do not have any idea why I did it this way... it seems like I should have just checked in a copy with git-lfs?

Reply:
The file *is* actually checked in (https://github.com/travitch/build-bom/blob/master/tests/sources/zlib.tar.gz), but something frequently deletes it and it has to be refreshed.  I haven't tracked down what is deleting it, but this is not usually causing much of a problem.